opensrp-reveal/src/main/java/org/smartregister/reveal/sync/LocationTaskIntentService.java

```diff
@@ -47,11 +44,33 @@ public void run() {
                 SyncServiceJob.scheduleJobImmediately(SyncServiceJob.TAG);
             }
         });
-        if (!syncedStructures.isEmpty() || !synchedTasks.isEmpty()) {
+
+        if (hasChangesInCurrentOperationalArea(syncedStructures, synchedTasks)) {
             Intent intent = new Intent(STRUCTURE_TASK_SYNCHED);
             LocalBroadcastManager.getInstance(getApplicationContext()).sendBroadcast(intent);
```

Comment:
Does this serve a different purpose than the broadcast here https://github.com/OpenSRP/opensrp-client-reveal/blob/21b831b741147ceffcc300c156e5b5655944fda7/opensrp-reveal/src/main/java/org/smartregister/reveal/sync/RevealClientProcessor.java#L95?

If so, please explain.

Reply:
Yes, Events and synced independenty from Tasks and Structures.

So we are broadcasting in 2 times
1.) When a new spray event is synced
2.) When a new structure or Task is synced

Reply:
Cool.